### PR TITLE
Active User Storage

### PR DESCRIPTION
**Pull request by thomasconner**
#### Description
Refactor to use the new active user storage API.

#### Changes
- Override the `Kinvey` class to use the custom `Client` class that stores the active user using keychain on iOS or Hawk on Android.
- Ignore all `.d.ts` files except `kinvey.d.ts` for npm.

### CODE REVIEW

File: src/kinvey.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  KinveyError,
+  CacheRequest,
+  RequestMethod
+} from 'kinvey-js-sdk/dist/export';
+import { Client } from './client';
+
+const USERS_NAMESPACE = 'user';
+const ACTIVE_USER_COLLECTION_NAME = 'kinvey_active_user';
+
+interface ClientConfig {
```
Comment (tejasranade) [inline src/kinvey.ts]: @thomasconner Does this have to reside in the shim? It seems like this could belong in core, there's nothing shim specific here.

Comment (thomasconner) [reply src/kinvey.ts]: I was getting a typescript error complaining that the config object passed to `init` did not have some properties. I agree that I would rather not have it here. For the meantime, I can remove it and mark the config object as type `any`.

Comment (tejasranade) [reply src/kinvey.ts]: Ok, you'll need to modify the `kinvey.d.ts` as well. 
As an alternative, what about adding a `ClientConfig` definition in `kinvey.d.ts`?

Comment (thomasconner) [reply src/kinvey.ts]: It is already in there.

**Review by tejasranade — DISMISSED** [review]

**Review by tejasranade — APPROVED** [review]

**Pull request merged**